python/ray/data/context.py

```diff
@@ -28,7 +28,7 @@
 DEFAULT_STREAMING_READ_BUFFER_SIZE = 32 * 1024 * 1024
 
 # Whether block splitting is on by default
-DEFAULT_BLOCK_SPLITTING_ENABLED = False
+DEFAULT_BLOCK_SPLITTING_ENABLED = True
```

Comment:
Turn on dynamic block splitting by default to test out CI tests.

Reply:
If there end up being a lot of follow-ups, we might want to flip this back off, merge this integration with it turned off, and open a dedicated follow-up PR for turning it on + corresponding fixes.

Reply:
yeah that's the plan.